src/main/java/gitflow/GitflowVersionTester.java

```diff
@@ -1,31 +1,58 @@
 package gitflow;
 
-import com.intellij.execution.configurations.GeneralCommandLine;
-import com.intellij.execution.ExecutionException;
-import com.intellij.execution.util.ExecUtil;
-import com.intellij.execution.process.ProcessOutput;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class GitflowVersionTester {
-	static Boolean isSupportedVersion = null;
 
-	static boolean isSupportedVersion(){
-		if (isSupportedVersion == null) {
+	private static final Logger logger = Logger.getInstance(GitflowVersionTester.class);
+
+	private static final Map<Project, GitflowVersionTester> testers = new ConcurrentHashMap<>();
+
+	public static GitflowVersionTester forProject(Project project) {
+		// Java 8
+//		return testers.computeIfAbsent(
+//			project,
+//			p -> new GitflowVersionTester(ServiceManager.getService(Gitflow.class), p)
+//		);
+		GitflowVersionTester ret = testers.get(project);
+		if (ret == null) {
+			ret = new GitflowVersionTester(ServiceManager.getService(Gitflow.class), project);
+			testers.put(project, ret);
```

Comment:
I think we should also handle removing on project close to prevent memory leaks in the map
Like here https://github.com/OpherV/gitflow4idea/blob/develop/src/main/java/gitflow/GitflowBranchUtilManager.java#L41

Reply:
Nifty!